src/disc/Albany_DiscretizationUtils.hpp

```diff
@@ -37,9 +38,50 @@ class SideStruct
   int      elem_ebIndex;   // index of element block that contains element
   unsigned side_local_id;  // local id of side relative to owning element
 };
-
 using SideSetList = std::map<std::string, std::vector<SideStruct>>;
 
+
+// This is a stucture that holds all of the sideset information over all worksets. When running populate mesh,
+//   there can be a huge number of worksets with only a handful (1-20) of entries each, which causes a huge
+//   performance hit on GPUs since each of these tiny views must be allocated serially on the GPU. These global/local
+//   side structs avoid this issue by allocating once after computing the global view extents and then providing
+//   local subviews to each workset.
+// Memory layout is enforced to be LayoutRight regardless of architecture since Kokkos parallel sections
+//   will only ever operate on the last extent of any of these Views and therefore they should be contiguous.
+class GlobalSideStruct
```

Comment:
I think the names of these structs are a bit misleading. Their name made me think they contain info on a single side, while instead they contain info of all sides in the current sideset (and global also spans all worksets). Maybe something like `GlobalSideSetInfo` and `LocalSideSetInfo`?

Also, on a "taste" note (i.e., feel free to ignore): I would not add `Struct` to a class name. We did it elsewhere in Albany, but I'm not a big fan of that.